src/pcre2_jit_compile.c

```diff
@@ -13772,6 +13772,49 @@ common->accept_label = LABEL();
 if (common->accept != NULL)
   set_jumps(common->accept, common->accept_label);
 
+/* Fail if we detect that the start position was moved to be either after
+the end position (\K in lookahead) or before the start offset (\K in
+lookbehind). */
+
+if (common->has_set_som &&
+    (common->re->extra_options & PCRE2_EXTRA_ALLOW_LOOKAROUND_BSK) == 0)
+  {
+  struct sljit_jump *bad_som;
+  struct sljit_jump *bad_eom;
+
+  // XXX emit code equivalent to the following:
+  // if (OVECTOR(0) < jit_arguments->str ||
+  //     OVECTOR(0) > OVECTOR(1))
+  //   {
+  //     return PCRE2_ERROR_BAD_BACKSLASH_K;
+  //   }
+
+  OP1(SLJIT_MOV, TMP1, 0, SLJIT_MEM1(SLJIT_SP), OVECTOR(0));
+  OP1(SLJIT_MOV, TMP2, 0, SLJIT_MEM1(SLJIT_SP), OVECTOR(1));
+  if (HAS_VIRTUAL_REGISTERS)
+    {
+    OP1(SLJIT_MOV, TMP3, 0, ARGUMENTS, 0);
+    OP1(SLJIT_MOV, TMP3, 0, SLJIT_MEM1(TMP3), SLJIT_OFFSETOF(jit_arguments, str));
+    }
+  else
+    {
+    OP1(SLJIT_MOV, TMP3, 0, SLJIT_MEM1(ARGUMENTS), SLJIT_OFFSETOF(jit_arguments, str));
+    }
+
+  // Compare if OVECTOR(0) < jit_arguments->str
+  bad_som = CMP(SLJIT_LESS, TMP1, 0, TMP3, 0);
+  // Compare if OVECTOR(0) > OVECTOR(1)
+  bad_eom = CMP(SLJIT_GREATER, TMP1, 0, TMP2, 0);
+
+  // If either comparison is true, return error and jump to abort
+  OP1(SLJIT_MOV, SLJIT_RETURN_REG, 0, SLJIT_IMM, PCRE2_ERROR_BAD_BACKSLASH_K);
+  JUMPTO(SLJIT_JUMP, common->abort_label);
+
+  // Patch the jumps to skip the error if the checks pass
+  JUMPHERE(bad_som);
+  JUMPHERE(bad_eom);
+  }
```

Comment:
The JIT is still very new to me; I don't really understand how it all works yet.

I've made an attempt at some code here. It crashes, but I'm trying.

Reply:
`sljit_emit_op0(compiler, SLJIT_BREAKPOINT);` stops gsb where you put it